src/analysis/normed/normed_field.lean

```diff
@@ -164,14 +164,42 @@ instance prod.non_unital_semi_normed_ring [non_unital_semi_normed_ring β] :
         ... = (∥x∥*∥y∥) : rfl,
   ..prod.semi_normed_group }
 
-/-- Seminormed group instance (using sup norm of sup norm) for matrices over a seminormed ring. Not
+lemma finset.sup_mul_le_mul_sup {α ι : Type*} [linear_ordered_semiring α] [order_bot α]
+  {a b : ι → α} (ha : 0 ≤ a) (hb : 0 ≤ b) (s : finset ι):
+  s.sup (a * b) ≤ s.sup a * s.sup b :=
+begin
+  apply finset.sup_le,
+  intros i hi,
+  classical,
+  rw [←finset.insert_erase hi, finset.sup_insert, finset.sup_insert, sup_eq_max, sup_eq_max,
+    pi.mul_apply],
+  refine mul_le_mul (le_max_left _ _) (le_max_left _ _) (hb i) (le_max_of_le_left $ ha i),
+end
```

Comment:
@YaelDillies, any suggestions for where to put this?

Reply:
Can it not go in `algebra.order.ring`? I believe `data.finset.lattice` is imported quite early on. Also, you're missing the `finset.inf`, `finset.sup'`, `finset.inf'`, nonpositive versions.

Reply:
Nope, it can't go there, that forms a cyclic import

Reply:
Ah yeah so `data.finset.lattice` works

Reply:
> nonpositive versions.

I skipped these because I'm not sure they make much sense in the context of `order_bot` (and because I didn't want to go too far down this rabbit hole)

Reply:
May you drop a TODO somewhere then?